packages/lesswrong/components/sequences/TopPostsPage.tsx

```diff
@@ -858,49 +852,98 @@ function TopSpotlightsSection({classes, yearGroupsInfo, sectionsInfo, reviewWinn
     filteredReviewWinnersForSpotlights = reviewWinnersWithPosts.filter(post => post.reviewWinner?.reviewYear === year && post.reviewWinner?.category === category)
   }
 
-  const filteredSpotlights = filterWhereFieldsNotNull(filteredReviewWinnersForSpotlights, 'spotlight').map(post => ({
-    ...post.spotlight,
-    post,
-    tag: null,
-    sequence: null,
-    ranking: post.reviewWinner?.reviewRanking
-  })).sort((a, b) => {
-    const reviewWinnerA = reviewWinnersWithPosts.find(post => post._id === a.post?._id)
-    const reviewWinnerB = reviewWinnersWithPosts.find(post => post._id === b.post?._id)
-    if (reviewWinnerA?.reviewWinner?.reviewRanking !== reviewWinnerB?.reviewWinner?.reviewRanking) {
-      return (reviewWinnerA?.reviewWinner?.reviewRanking ?? 0) - (reviewWinnerB?.reviewWinner?.reviewRanking ?? 0)
-    } else if (reviewWinnerA?.reviewWinner?.reviewYear !== reviewWinnerB?.reviewWinner?.reviewYear) {  
-      return (reviewWinnerA?.reviewWinner?.reviewYear ?? 0) - (reviewWinnerB?.reviewWinner?.reviewYear ?? 0)
-    } else if (reviewWinnerA?.reviewWinner?.category !== reviewWinnerB?.reviewWinner?.category) {
-      return (reviewWinnerA?.reviewWinner?.category ?? '').localeCompare(reviewWinnerB?.reviewWinner?.category ?? '')
-    } else {
-      return 0
+  const postsResult = useQuery(gql(`
+    query GetReviewWinnerSpotlights($selector: PostSelector, $limit: Int, $enableTotal: Boolean) {
+      posts(selector: $selector, limit: $limit, enableTotal: $enableTotal) {
+        results {
+          ...PostsBase
```

Comment:
Do we need to be querying and refetching the posts (and reviewWinners) themselves?  We should already have them from `reviewWinnersWithPosts`, no?  (If this is because there's no convenient way to query the spotlights based on postId, that can probably just be a view?)

Reply:
(or if we want to be really lazy, we can at least drop `PostsBase` from the result, just return the post _ids, and then merge them with the `reviewWinnersWithPosts` by _id? unless that'll mess up the apollo cache, in which case that might be a risk with getting `PostsBase` instead of `PostsTopItemInfo`?)

Reply:
The first query, which gets all review-winners without any filtering, has a narrow fragment which doesn't include everything needed for a spotlight item (in particular not content excerpts, but also missing other things). This query runs on only the posts that matched the filters, with more inclusive fragments. There is still some overlap that could be eliminated, but it's not as simple as joining the queries on post IDs. (This component used to do that, this changeset moves away from that, because of the fragment size)